src/java.base/share/classes/java/util/concurrent/ForkJoinPool.java

```diff
@@ -1338,8 +1341,8 @@ private ForkJoinTask<?> nextLocalTask(int fifo) {
             ForkJoinTask<?> t = null;
             ForkJoinTask<?>[] a = array;
             int b = base, p = top, cap;
-            if (a != null && (cap = a.length) > 0) {
-                for (int m = cap - 1, s, nb; p - b > 0; ) {
+            if (p - b > 0 && a != null && (cap = a.length) > 0) {
+                for (int m = cap - 1, s, nb;;) {
```

Comment:
@DougLea I'm curious, what effect did you see if moving the `p - b > 0` out of the loop conditional?

Reply:
This was part of saving a read in the common case of empty local queue. Barely but reliably worth doing.

Reply:
👍